src/transformers/models/vits/convert_original_checkpoint.py

```diff
@@ -0,0 +1,384 @@
+                    layer_index = name.split(key)[0].split(".")[-2]
+
+                    # remap the layer index since we removed the Flip layers
+                    if "flow.flows" in mapped_key:
+                        layer_index = str(int(layer_index) // 2)
+                    if "duration_predictor.flows" in mapped_key or "duration_predictor.post_flows" in mapped_key:
+                        layer_index = str(int(layer_index) // 2 + 1)
+
+                    mapped_key = mapped_key.replace("*", layer_index)
+                if "weight_g" in name:
+                    weight_type = "weight_g"
+                elif "weight_v" in name:
+                    weight_type = "weight_v"
+                elif "bias" in name:
+                    weight_type = "bias"
+                elif "weight" in name:
+                    weight_type = "weight"
+                elif "running_mean" in name:
+                    weight_type = "running_mean"
+                elif "running_var" in name:
+                    weight_type = "running_var"
+                elif "num_batches_tracked" in name:
+                    weight_type = "num_batches_tracked"
+                else:
+                    weight_type = None
+                set_recursively(hf_model, mapped_key, value, name, weight_type)
+            continue
+        if not is_used:
+            unused_weights.append(name)
+
+    logger.warning(f"Unused weights: {unused_weights}")
+
+
+@torch.no_grad()
+def convert_checkpoint(
+    pytorch_dump_folder_path,
+    checkpoint_path=None,
+    config_path=None,
+    vocab_path=None,
+    language=None,
+    num_speakers=None,
+    repo_id=None,
+):
+    """
+    Copy/paste/tweak model's weights to transformers design.
+    """
+    if config_path is not None:
+        config = VitsConfig.from_pretrained(config_path)
+    else:
+        config = VitsConfig()
+
+    if num_speakers:
+        config.num_speakers = num_speakers
+        config.speaker_embedding_size = 256
+
+    if checkpoint_path is None:
+        logger.info(f"***Converting model: facebook/mms-tts {language}***")
+
+        vocab_path = hf_hub_download(
+            repo_id="facebook/mms-tts",
+            filename="vocab.txt",
+            subfolder=f"models/{language}",
+        )
+        config_file = hf_hub_download(
+            repo_id="facebook/mms-tts",
+            filename="config.json",
+            subfolder=f"models/{language}",
+        )
+        checkpoint_path = hf_hub_download(
+            repo_id="facebook/mms-tts",
+            filename="G_100000.pth",
+            subfolder=f"models/{language}",
+        )
+
+        with open(config_file, "r") as f:
+            data = f.read()
+            hps = json.loads(data)
+
+        is_uroman = hps["data"]["training_files"].split(".")[-1] == "uroman"
+        if is_uroman:
+            logger.warning("For this checkpoint, you should use `uroman` to convert input text before tokenizing it!")
```

Comment:
Took a deeper look into this - the normaliser they use is a `perl` based normaliser that converts non-Roman alphabets to Roman: https://github.com/isi-nlp/uroman/blob/master/lib/NLP/Romanizer.pm

This means that we can't simply add it as a soft dependency - it's not a Python package, but instead a `perl` package that requires a `perl` execution on the command line to run this pre-processing

Now this leaves us with two options:
1. Port the `perl`-based normaliser to Python to use in our tokenizer file: https://github.com/isi-nlp/uroman/blob/master/lib/NLP/Romanizer.pm
2. Throw a warning when a user uses a language that requires the normaliser, encouraging them to pre-process using this `perl` normaliser before using the tokenizer

Advantage of 1 is that it's a cleaner user experience - they don't have to worry about running any extra pre-processing. Disadvantage is that it would take a day or two to port

Happy to take a look into porting the code, but it might be faster to first release without the normaliser, and then add it later if we see usage for the non-roman checkpoints?

Reply:
In this case, it sounds like a lot of work to add it, so I'd say no. @ArthurZucker would it still be keeping with tokenizer patterns if we add decided to add later on with an optional flag e.g. do_uroman_normalization?

Could you add this info to docs/source/en/model_doc/vits.md so that someone who uses the model knows they need to do this normalization themselves (possibly with a code example?)

Reply:
Think there's a nice middle ground here where the user can clone the `uroman` package:

```
git clone https://github.com/isi-nlp/uroman.git
```

Then specify the path to the package when they call the `tokenizer`:
```python
inputs = tokenizer(text="Πώς είσαι", uroman_path="./uroman", return_tensors="pt")
```

Under the hood, the `tokenizer` executes the correct PERL command to pre-process the text inputs. This way, the user doesn't have to do an extra pre-processing step outside model. But it isn't ideal since they do need to pass the path to the `uroman` package each time they call the tokenizer

Reply:
Sorry, for the delayed reply, I missed this comment earlier. 

Let's not do this. Running subprocesses from python should be avoided as much as possible. It's potentially dangerous, especially if we're allowing users to pass in arbitrary paths and is a good indication that something should be done outside of the python logic. 

I suggest just checking if the input text has roman letters. If not, we raise either a warning or an exception telling the user to run the command to process their data using `uroman`.

Reply:
Sure sounds good - updated the tokenizer to reflect these changes and wrote a section in the docs: https://github.com/huggingface/transformers/pull/24085/commits/90679930110a7ceec07b29a5bfb973ebf0510602